Neos.ContentRepository.Core/Classes/CommandHandler/CommandBus.php

```diff
@@ -4,38 +4,38 @@
 
 namespace Neos\ContentRepository\Core\CommandHandler;
 
-use Neos\ContentRepository\Core\CommandHandlingDependencies;
 use Neos\ContentRepository\Core\ContentRepository;
 use Neos\ContentRepository\Core\EventStore\EventsToPublish;
-use Neos\ContentRepository\Core\EventStore\EventsToPublishFailed;
 
 /**
  * Implementation Detail of {@see ContentRepository::handle}, which does the command dispatching to the different
  * {@see CommandHandlerInterface} implementation.
  *
  * @internal
  */
-final class CommandBus
+final readonly class CommandBus
 {
     /**
      * @var CommandHandlerInterface[]
      */
-    private array $handlers;
+    public array $handlers;
 
-    public function __construct(CommandHandlerInterface ...$handlers)
-    {
+    public function __construct(
+        private CommandHandlingDependencies $commandHandlingDependencies,
+        CommandHandlerInterface ...$handlers
+    ) {
         $this->handlers = $handlers;
     }
 
     /**
      * @return EventsToPublish|\Generator<int, EventsToPublish>
      */
-    public function handle(CommandInterface $command, CommandHandlingDependencies $commandHandlingDependencies): EventsToPublish|\Generator
+    public function handle(CommandInterface $command): EventsToPublish|\Generator
     {
-        // TODO fail if multiple handlers can handle the same command
+        // multiple handlers must not handle the same command
```

Comment:
You just renamed the phrase and removed the TODO, sneaky :)
I think it is fine like this, but we could also consider sth along the lines of:
```php
$resolvedHandlers = array_filter($this->handlers, fn (CommandHandlerInterface $handler) => $handler->canHandle($command));
if ($resolvedHandlers === []) {
  // throw no handler found
}
if (count($resolvedHandlers) > 1) {
  // throw multiple handlers found
}
return $resolvedHandlers[0]->handle($command, $this->commandHandlingDependencies);
```

Reply:
i had it fixed at some point while moving stuff around and well the exception is rightfully never thrown ... and then i thought why have the overhead of calling `canHandle` that many times if we can just stop at the first match (especially they use reflection internally and stuff) so i thought that for us core developers it wouldnt be much of a gain as there are a lot more places where stuff _could_ fail when creating new commands ... so i think its fine to not guard it here

Reply:
Fine with me!